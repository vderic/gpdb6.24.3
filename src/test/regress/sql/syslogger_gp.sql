-- Smoke tests for syslogger (since GP syslogger code is divergent from upstream)

create or replace function log_and_retrieve_master(log_message text)
returns table (log_msg text) as
$$
declare
  session_id text;
begin
  raise log '%', log_message;

  -- We need to check only latest logs (otherwise we will get all the logs
  -- including logs generated by previous callings since we cannot delete
  -- previous logs)
  return query select logmessage from gp_toolkit.gp_log_master_concise where logmessage = log_message order by logtime desc limit 1;
end;
$$ language plpgsql;

select log_and_retrieve_master('message with a " mark');
select log_and_retrieve_master('message with two "" mark');
select log_and_retrieve_master('message with three """ mark');
select log_and_retrieve_master('message with foo"');
select log_and_retrieve_master('message with ""foo""bar"');
select log_and_retrieve_master('message with no quotes');
