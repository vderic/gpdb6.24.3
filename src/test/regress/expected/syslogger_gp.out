-- Smoke tests for syslogger (since GP syslogger code is divergent from upstream)
create or replace function log_and_retrieve_master(log_message text)
returns table (log_msg text) as
$$
declare
  session_id text;
begin
  raise log '%', log_message;

  -- We need to check only latest logs (otherwise we will get all the logs
  -- including logs generated by previous callings since we cannot delete
  -- previous logs)
  return query select logmessage from gp_toolkit.gp_log_master_concise where logmessage = log_message order by logtime desc limit 1;
end;
$$ language plpgsql;
select log_and_retrieve_master('message with a " mark');
 log_and_retrieve_master 
-------------------------
 message with a " mark
(1 row)

select log_and_retrieve_master('message with two "" mark');
 log_and_retrieve_master  
--------------------------
 message with two "" mark
(1 row)

select log_and_retrieve_master('message with three """ mark');
   log_and_retrieve_master   
-----------------------------
 message with three """ mark
(1 row)

select log_and_retrieve_master('message with foo"');
 log_and_retrieve_master 
-------------------------
 message with foo"
(1 row)

select log_and_retrieve_master('message with ""foo""bar"');
 log_and_retrieve_master  
--------------------------
 message with ""foo""bar"
(1 row)

select log_and_retrieve_master('message with no quotes');
 log_and_retrieve_master 
-------------------------
 message with no quotes
(1 row)

